plugin/background.js

```diff
@@ -0,0 +1,44 @@
+function getActiveTab() {
+    return new Promise(function (resolve, reject) {
+        chrome.tabs.query({currentWindow: true, active: true}, function (tabs) {
+            if ((tabs != null) && (0 < tabs.length)) {
+                resolve(tabs[0].id);
+            } else {
+                reject();
+            }
+        });
+    });
+}
+
+function openTabWindow() {
+    // open new tab window, passing ID of open tab with content to convert to epub as query parameter.
+    getActiveTab().then(function (tabId) {
+        let url = chrome.runtime.getURL("popup.html") + "?id=";
+        url += tabId;
+        chrome.tabs.create({url: url});
+        window.close();
+    });
+}
+
+chrome.browserAction.onClicked.addListener(
+    function () {
+        openTabWindow();
+    }
+);
+
+chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
```

Comment:
Used to have a runtime sendmessage in the downloads that wrapped the fetch call to go through here and return the response. The header ended up being null everytime.